src/pools/balancer/Pool.ts

```diff
@@ -706,16 +708,18 @@ export class Pool {
   }
 
   /**
-   * updateMarketFeeCollector - updates marketFeeCollector - can be called only by the marketFeeCollector
+   * updatePublishMarketFee - updates marketFeeCollector - can be called only by the marketFeeCollector
    * @param {String} address
    * @param {String} poolAddress
    * @param {String} newCollector new market fee collector address
+   * @param {String} newPublishMarketSwapFee new market swap fee
    * @return {TransactionReceipt}
    */
-  async updateMarketFeeCollector(
+  async updatePublishMarketFee(
     address: string,
     poolAddress: string,
-    newCollector: string
+    newCollector: string,
+    newPublishMarketSwapFee?: string
```

Comment:
This method is used for updating publishMarketSwapFee so why would this here be optional? And then magically replaced with a value further down if not passed?